### PR TITLE
Sync with Siddhi 5.0.0 API changes

### PR DESCRIPTION
**Pull request by ajkarki**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/LengthTimeLinearRegressionForecastStreamProcessor.java [hunk 1]
```diff
@@ -104,29 +107,32 @@
...
     private static final int SIMPLE_LINREG_INPUT_PARAM_COUNT = 2; //Number of input parameters in
     private int paramCount; // Number of x variables +1
     private long duration; // Time window to consider for regression calculation
     private int calcInterval = 1; // The frequency of regression calculation
     private double ci = 0.95; // Confidence Interval simple linear forecast
     private LengthTimeRegressionCalculator regressionCalculator = null;
+    private List<Attribute> attributes;
     private int yParameterPosition;
-    // simple linear regression
+    // simple linear regressionmvn
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/LengthTimeLinearRegressionForecastStreamProcessor.java]: regressionmvn  ??

File: pom.xml [hunk 2]
```diff
@@ -199,7 +199,7 @@
                 <plugin>
                     <groupId>org.apache.felix</groupId>
                     <artifactId>maven-bundle-plugin</artifactId>
-                    <version>2.3.5</version>
+                    <version>4.2.0</version>
```
Comment (mohanvive) [inline pom.xml]: Use 3.x.x version.

File: component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/extrema/KalmanMinMaxStreamProcessor.java [hunk 3]
```diff
@@ -189,19 +191,21 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
                 streamEventChunk.remove();
                 Double eventKey = (Double) event.getAttribute(variablePosition);
                 extremaCalculator = new ExtremaCalculator(q, r);
-                eventStack.add(event);
-                valueStack.add(eventKey);
+                extensionState.addInEventStack(event);
+                extensionState.addInValueStack(eventKey);
+                
+
```
Comment (mohanvive) [inline component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/extrema/KalmanMinMaxStreamProcessor.java]: Remove unnecessary line spaces. Check other places as well.

Comment (ajkarki) [reply component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/extrema/KalmanMinMaxStreamProcessor.java]: Fix with 957112b

File: pom.xml (hunk 2 above)
Comment (ajkarki) [reply pom.xml]: Fix with 957112b

File: component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/LengthTimeLinearRegressionForecastStreamProcessor.java (hunk 1 above)
Comment (ajkarki) [reply component/src/main/java/org/wso2/extension/siddhi/execution/timeseries/LengthTimeLinearRegressionForecastStreamProcessor.java]: Fix with 957112b

**Pull request merged**